localstack-core/localstack/packages/java.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+from typing import List
+
+from localstack.packages import Package
+from localstack.packages.core import ArchiveDownloadAndExtractInstaller
+from localstack.utils.platform import Arch, get_arch, is_linux, is_mac_os
+
+LOG = logging.getLogger(__name__)
+
+# Default version if not specified
+DEFAULT_JAVA_VERSION = "11"
+
+# Supported Java LTS versions mapped with Eclipse Temurin build semvers
+JAVA_VERSIONS = {
+    "8": "8u422-b05",
+    "11": "11.0.24+8",
+    "17": "17.0.12+7",
+    "21": "21.0.4+7",
+}
+
+JRE_DISTRIB_URL = "https://github.com/adoptium/temurin{version}-binaries/releases/download/{tag_slug}/OpenJDK{version}U-jre_{arch}_{os}_hotspot_{semver_safe}.tar.gz"
+
+
+class JavaPackageInstaller(ArchiveDownloadAndExtractInstaller):
```

Comment:
nit: Since this package installer is introduced here, and might be removed from the `Dockerfile` in the future, it might make sense to call this installer in the `_prepare_installation` steps of installers which depend on Java (but we can also do that in a later iteration when we actually want to get rid of the JRE).